### PR TITLE
Fix #62: Implement async job execution

### LINKED ISSUE #62: Asynchronous code
I have trouble getting asynchronous code (async/await, Promise's) to work. I tried several approaches without much luck:

```python
ctx = quickjs.Context()

ctx.add_callable("print", print)

ctx.eval("""
async function a() {
    print(1 + await 1);
}

a();
""")
```

```python
ctx.eval("""
Promise.resolve(3).then(
    r => print(r)
);
""")
```

```python
f = quickjs.Function("f", """
var b = {};

async function a() {
    b.x = 1 + await 1;
}

a()

function f() {
    while (!b.x) {}
    return b.x;
}
""")

f()
```

The two first codes execute without printing anything. The third code hangs due to non-terminating busy waiting.

Note: the codes work correctly in `qjs` (quickjs' interactive prompt), of course replacing `print` with `console.log` in the first two examples.

### PR DESCRIPTION
**Pull request by qwenger**
Fix #62.

Note that errors occurring in asynchronous code do not seem to be raised (even though the case `JS_ExecutePendingJob(...) < 0` is handled). But this seems to be the case in the native `qjs` as well.
EDIT: this seems to be a matter of using `JS_SetHostPromiseRejectionTracker`, see https://github.com/bellard/quickjs/issues/87. Should we use it, or maybe conditionally with a flag on the context (the question also applies to #64 ...)?

### CODE REVIEW

**Comment by qwenger** [comment]
Note: with this patch, in the presence of potentially asynchronous code, the standard thing would be to call `execute_pending_job` repeatedly:

```python
ctx.eval("/* some job launching */")
while ctx.execute_pending_job():
    pass
ctx.eval("/* some more code */")
```

If this back-and-forth between Python and JS is to slow, we could implement a `execute_all_pending_jobs` method with a loop on the C side, just like `js_std_loop` (https://github.com/bellard/quickjs/blob/master/quickjs-libc.c#L3878).

File: module.c [hunk 1]
```diff
@@ -379,6 +383,26 @@ static PyObject *context_module(ContextData *self, PyObject *args) {
 	return context_eval_internal(self, args, JS_EVAL_TYPE_MODULE);
 }
 
+// _quickjs.Context.execute_pending_job
+//
+// If there are pending jobs, executes one and returns True. Else returns False.
+static PyObject *context_execute_pending_job(ContextData *self) {
+	JSContext *ctx;
+	JSValue value;
```
Comment (PetterS) [inline module.c:391]: Unused.

File: module.c [hunk 2]
```diff
@@ -379,6 +383,26 @@ static PyObject *context_module(ContextData *self, PyObject *args) {
...
+	int ret;
+	prepare_call_js(self);
+	ret = JS_ExecutePendingJob(self->runtime, &ctx);
```
Comment (PetterS) [inline module.c:394]: This file uses C99 where variables may be declared where they are first assigned to.

**Review by PetterS — APPROVED** [review]
Looks good! 

This should be useful

**Pull request merged**